src/tests/enzyme_smoke.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Tribol Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (MIT)
+
+//-----------------------------------------------------------------------------
+//
+// file: enzyme_smoke.cpp
+//
+//-----------------------------------------------------------------------------
+
+#include "gtest/gtest.h"
+
+template <typename return_type, typename... Args>
+return_type __enzyme_fwddiff( Args... );
+
+void LinearQuadBasis( const double* xi, double* phi )
+{
+  phi[0] = 0.25 * ( 1 - xi[0] ) * ( 1 - xi[1] );
+  phi[1] = 0.25 * ( 1 + xi[0] ) * ( 1 - xi[1] );
+  phi[2] = 0.25 * ( 1 + xi[0] ) * ( 1 + xi[1] );
+  phi[3] = 0.25 * ( 1 - xi[0] ) * ( 1 + xi[1] );
+}
+
+void LinearQuadBasisDeriv( const double* xi, double* phi, double* dphi_dxi, double* dphi_deta )
+{
+  double xi_dot[2] = { 1.0, 0.0 };
+  __enzyme_fwddiff<void>( (void*)LinearQuadBasis, xi, xi_dot, phi, dphi_dxi );
+  xi_dot[0] = 0.0;
+  xi_dot[1] = 1.0;
+  __enzyme_fwddiff<void>( (void*)LinearQuadBasis, xi, xi_dot, phi, dphi_deta );
+}
+
+TEST( enzyme_smoke, basic_use )
+{
+  double xi[2] = { 0.2, -0.4 };
```

Comment:
Anything special about this point in parent space? Or is it random?

Reply:
Random